ui.js

```diff
@@ -43,27 +51,123 @@ var PopupBluetoothDeviceMenuItem = GObject.registerClass(
                 this.add(this._statusBin, { expand: false });
             }
 
+            this._bat_icon = new St.Icon({ style_class: 'popup-menu-icon' });
+            this.insert_child_at_index(this._bat_icon, this.get_n_children() - 1);
+            this._bat_icon.icon_name = null;
+
+            // dirty trick: instantiate the label with text 100%, so we can set
+            // the natural width of the label in case monospace has no effect
+            this._bat_percentage_label = new St.Label( {y_expand: false, style_class: 'monospace', text: '100%'} );
+            this._bat_percentage_label.natural_width = this._bat_percentage_label.width;
+            this._bat_percentage_label.text = "";
+
+            this._bat_percentage_label.x_expand = false;
+            this._bat_percentage_label.x_align = Clutter.ActorAlign.LEFT;
+            this.insert_child_at_index(this._bat_percentage_label, this.get_n_children() - 1);
```

Comment:
It would be nice to create new class (eg. `BatteryLabel`) with all this fields and functions like `setBatteryLevel` and `clear`.

Reply:
Done